pkg/metrics/metrics.go

```diff
@@ -0,0 +1,126 @@
+	}, []string{"name", "namespace", "uid", "condition", "status"})
+
+	MachineSetStatusFailedMachines = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_set_failed_machines",
+		Help: "Information of the mcm managed Machinesets' failed machines.",
+	}, []string{"name", "namespace", "uid", "failed_Machine_name", "failed_machine_provider_id", "failed_machine_owner_ref",
+		"failed_Machine_last_operation_description", "failed_machine_last_operation_last_update_time", "failed_machine_last_operation_state",
+		"failed_Machine_last_operation_machine_operation_type"})
+
+	MachineSetStatus = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_set_status",
+		Help: "Information of the mcm managed Machinesets' status conditions.",
+	}, []string{"name", "namespace", "uid", "available_replicas",
+		"fully_labeled_replicas", "ready_replicas", "replicas"})
+
+	MachineDeploymentCountDesc = prometheus.NewDesc("mcm_machinedeployment_items_total", "Count of machinedeployments currently managed by the mcm.", nil, nil)
+	MachineDeploymentCreated   = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_deployment_created",
+		Help: "Creation time of the Machinedeployments currently managed by the mcm.",
+	}, []string{"name", "namespace", "uid"})
+	MachineDeploymentInfo = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_deployment_info",
+		Help: "Information of the Machinedeployments currently managed by the mcm.",
+	}, []string{"name", "namespace", "uid", "generation", "kind", "api_version", "spec_replicas", "spec_strategy_type",
+		"spec_paused", "spec_revision_history_limit", "spec_progress_deadline_seconds", "spec_min_ready_seconds", "spec_rollbackto_revision",
+		"spec_strategy_rolling_update_max_surge", "spec_strategy_rolling_update_max_unavailable"})
+
+	MachineDeploymentStatusCondition = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_deployment_status_condition",
+		Help: "Information of the mcm managed Machinedeployments' status conditions.",
+	}, []string{"name", "namespace", "uid", "condition", "status"})
+
+	MachineDeploymentStatus = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_deployment_status",
+		Help: "Information of the mcm managed Machinedeployments' status conditions.",
+	}, []string{"name", "namespace", "uid", "available_replicas", "unavailable_replicas", "ready_replicas",
+		"updated_replicas", "collision_count", "replicas"})
+	MachineDeploymentStatusFailedMachines = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mcm_machine_deployment_failed_machines",
+		Help: "Information of the mcm managed Machinedeployments' failed machines.",
+	}, []string{"name", "namespace", "uid", "failed_Machine_name", "failed_machine_provider_id", "failed_machine_owner_ref",
+		"failed_Machine_last_operation_description", "failed_machine_last_operation_last_update_time", "failed_machine_last_operation_state",
```

Comment:
Could we write the label names completely in lowercase?

Reply:
Fixed.